spec/controllers/hq/sponsorships_controller_spec.rb

```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.describe Hq::SponsorshipsController, type: :controller do
+  let(:sponsorship) {create :sponsorship, active: true}
+
+  before :each do
+    sign_in instance_double(AdminUser)
+  end
+
+  specify "#inactivate" do
+    expect(Sponsorship).to receive(:find).and_return(sponsorship)
+    put :inactivate, id: sponsorship.id, sponsorship: {end_date: Date.current}
+
+    expect(sponsorship.active).to be false
```

Comment:
you can't stub `update!` because Rails `sponsorship.update!(active: false, end_date: end_date)` also changes `sponsorship.active` to `false`.

If you do:
```ruby
    expect(Sponsorship).to receive(:find).and_return(sponsorship)
    expect(sponsorship).to receive(:update!).and_return(true)
```
it will not change `sponsorship.active` to `false` and line 14 will fail.

And if you do:
```ruby
    expect(Sponsorship).to receive(:find).and_return(sponsorship)
    sponsorship.active = false
    expect(sponsorship).to receive(:update!).and_return(sponsorship)
```
then `:find` will also return a sponsorship with `sponsorship.active = false` and then the test has no value. ( because then we could change `sponsorship.update!(active: false, end_date: end_date)` to `sponsorship.update!(  end_date: end_date)` and the test would still pass)